Modules/ThirdParty/TIFF/src/itktiff/CMakeLists.txt

```diff
@@ -1 +1,254 @@
+# EXPRESS, IMPLIED OR OTHERWISE, INCLUDING WITHOUT LIMITATION, ANY
+# WARRANTY OF MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE.
+#
+# IN NO EVENT SHALL SAM LEFFLER OR SILICON GRAPHICS BE LIABLE FOR
+# ANY SPECIAL, INCIDENTAL, INDIRECT OR CONSEQUENTIAL DAMAGES OF ANY KIND,
+# OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+# WHETHER OR NOT ADVISED OF THE POSSIBILITY OF DAMAGE, AND ON ANY THEORY OF
+# LIABILITY, ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE
+# OF THIS SOFTWARE.
+
+if(FALSE) # XXX(ITK): ITK handles CMake versions.
+cmake_minimum_required(VERSION 3.9.0)
+
+# Default policy is from 3.9.0
+cmake_policy(VERSION 3.9.0)
+endif()
+
+# Find CMake modules in cmake/
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake")
+
+if(FALSE) # XXX(ITK): We hard-code the TIFF version that we imported.
+# Read version information from configure.ac.
+include(AutotoolsVersion)
+message(STATUS "Building tiff version ${LIBTIFF_VERSION_FULL}")
+else()
+# Hard-code the version of TIFF used by ITK.
+set(LIBTIFF_MAJOR_VERSION 4)
+set(LIBTIFF_MINOR_VERSION 7)
+set(LIBTIFF_MICRO_VERSION 0)
+set(LIBTIFF_ALPHA_VERSION)
+set(LIBTIFF_VERSION "${LIBTIFF_MAJOR_VERSION}.${LIBTIFF_MINOR_VERSION}.${LIBTIFF_MICRO_VERSION}")
+set(LIBTIFF_VERSION_FULL "${LIBTIFF_VERSION}${LIBTIFF_ALPHA_VERSION}")
+endif()
+
+# Project definition
+set(CMAKE_C_STANDARD 99)
+project(tiff
+        VERSION "${LIBTIFF_VERSION}"
+        LANGUAGES C CXX)
+
+if(FALSE) # XXX(ITK): Suppress options not relevant to ITK.
+if(NOT CMAKE_SOURCE_DIR STREQUAL PROJECT_SOURCE_DIR)
+  # libtiff is being included as a subproject of some other project.
+  set(TIFF_INSTALL_DEFAULT OFF)
+else()
+  set(TIFF_INSTALL_DEFAULT ON)
+  option(BUILD_SHARED_LIBS "Build shared libraries" ON)
+endif()
+
+option(tiff-tools "build TIFF tools" ON)
+option(tiff-tests "build TIFF tests" ON)
+option(tiff-contrib "build TIFF contrib" ON)
+option(tiff-docs "build TIFF documentation" ON)
+option(tiff-deprecated "build TIFF deprecated features" OFF)
+option(tiff-install "install TIFF targets" ${TIFF_INSTALL_DEFAULT})
+else()
+set(tiff-deprecated ON)
+set(tiff-install OFF)
+endif()
+
+# Disable deprecated features to ensure clean build
+if (tiff-deprecated)
+  add_definitions(-DTIFF_DISABLE_DEPRECATED)
+endif()
+
+# Autotools compatibility
+include(AutotoolsCompat)
+
+# Compiler checks (warning flags, etc.)
+include(CompilerChecks)
+
+# Linker checks (version script, etc.)
+include(LinkerChecks)
+
+# Processor checks (endianness, fill order, floating point, etc.)
+include(ProcessorChecks)
+
+# Checks for headers
+include(IncludeChecks)
+
+# Checks for functions and other symbols
+include(SymbolChecks)
+
+# Standard installation paths
+include(GNUInstallDirs)
+# Documentation install directory (default to cmake project docdir)
+set(LIBTIFF_DOCDIR "${CMAKE_INSTALL_DOCDIR}")
+
+if(FALSE) # XXX(ITK): ITK handles testing
+# CTest testing
+enable_testing()
+endif()
```

Comment:
I guess this `if(FALSE) ... endif()` may be removed, right?

Reply:
The idea is to have two new lines that suppress logic in between them, instead of removing blocks of code, in order to make it easier to resolve conflicts when updating from upstream in the future.

Reply:
You can find the documentation for updating third party libraries here: 
https://github.com/InsightSoftwareConsortium/ITK/blob/master/Documentation/docs/contributing/updating_third_party.md
This should give you an overview of the process.

Reply:
Thanks @blowekamp Do I understand correctly that two out of the five commits of this PR are scripted? So when reviewing, the focus should mostly be on the three "manual" commits, right? (The ones that are labeled "ENH".)

Reply:
The commit titled "tiff: 2024-09-11" is generated by the script as a snapshot of the upstream version.  The merge commit was scripted as a subtree merge, though some conflict resolution was done manually.  Yes, the other three commits are the manual steps.